content-src/components/DiscoveryStreamBase/DiscoveryStreamBase.jsx

```diff
@@ -152,49 +194,21 @@ export class _DiscoveryStreamBase extends React.PureComponent {
     }
   }
 
-  renderStyles(styles) {
-    // Use json string as both the key and styles to render so React knows when
-    // to unmount and mount a new instance for new styles.
-    const json = JSON.stringify(styles);
-    return (<style key={json} data-styles={json} ref={this.onStyleMount} />);
-  }
-
-  componentWillReceiveProps(oldProps) {
-    if (this.props.DiscoveryStream.layout !== oldProps.DiscoveryStream.layout) {
-      rickRollCache = [];
-    }
-  }
-
-  render() {
-    // Select layout render data by adding spocs and position to recommendations
-    const layoutRender = selectLayoutRender(this.props.DiscoveryStream, rickRollCache);
-    const styles = [];
-    const {spocs, feeds} = this.props.DiscoveryStream;
-
-    if (!spocs.loaded || !feeds.loaded) {
-      return null;
-    }
-
-    return (
-      <div className="discovery-stream ds-layout">
-        {layoutRender.map((row, rowIndex) => (
-          <div key={`row-${rowIndex}`} className={`ds-column ds-column-${row.width}`}>
-            <div className="ds-column-grid">
-              {row.components.map((component, componentIndex) => {
-                styles[rowIndex] = [...styles[rowIndex] || [], component.styles];
-                return (<div key={`component-${componentIndex}`}>
-                  {this.renderComponent(component, row.width)}
-                </div>);
-              })}
-            </div>
-          </div>
-        ))}
-        {this.renderStyles(styles)}
-      </div>
-    );
-  }
-}
+  return (
+  <div {...props} className={`ds-column ds-column-${props.row.width}`}>
```

Comment:
What prop are we trying to get from here? I believe we want to avoid spread on component properties.

Reply:
+1,  its redundant